src/mono/mono/mini/aot-runtime.c

```diff
@@ -3704,10 +3706,10 @@ decode_patch (MonoAotModule *aot_module, MonoMemPool *mp, MonoJumpInfo *ji, guin
 			if (ref.method) {
 				ji->data.method = ref.method;
 			}else {
-				ERROR_DECL (error);
 				ji->data.method = mono_get_method_checked (ref.image, ref.token, NULL, NULL, error);
 				if (!ji->data.method)
 					g_error ("AOT Runtime could not load method due to %s", mono_error_get_message (error)); /* FIXME don't swallow the error */
+				mono_error_cleanup (error);
```

Comment:
This doesn't seem right. You might need to re-init `error`. (or is this really just to appease the compiler?  if `error` only has content if`!ji->data.method`, in which case we call `g_error` and abort?)

Why not use a `ERROR_DECL (local_error)` instead?

Reply:
You have the same pattern in several parts of this method, and you can reuse the error as long as it doesn't carry an error without re-init it and then it won't be reused since it will break out of switch statement and return out from function, but I agree that we should end up hitting g_error (that could abort depending on used logger) or the g_assert, so doing the cleanup might not really be needed. Normally we assert on errors or cleanup, but since this code path should get an error if it returns null and that we apparently see it as a critical error, maybe it is better to replace the condition and g_error with a mono_error_assert_msg_ok.